src/dowel/csv_output.py

```diff
@@ -48,7 +50,35 @@ def record(self, data, prefix=''):
                            'Did you change key sets after your first '
                            'logger.log(TabularInput)?'.format(
                                set(self._fieldnames), set(to_csv.keys())))
-
+                
+                # close any existing log file
+                super().close()
+
+                # Read data from the existing log file once
+                with open(self._filename, 'r') as existing_file:
+                    reader = csv.DictReader(existing_file)
+                    file_data = []
+                    for row in reader:
+                        file_data.append(row)
```

Comment:
This loop can actually all just be
```python
file_data = list(csv.DictReader(existing_file))
```

The comment is good to have to, since it's important that this data is read before we over-write it.

Reply:
Thanks for suggesting a more efficient way for the code pieces